railties/lib/rails/application/configuration.rb

```diff
@@ -577,10 +577,12 @@ def initialize
         def method_missing(method, *args)
           if method.end_with?("=")
             @configurations[:"#{method[0..-2]}"] = args.first
-          else
+          elsif args.none?
             @configurations.fetch(method) {
               @configurations[method] = ActiveSupport::OrderedOptions.new
             }
+          else
+            super
```

Comment:
Delegating to `super` for fallback behavior will raise a `NoMethodError`. While the outcome is helpful (an error instead of silent misbehavior), I wonder if raising an ad-hoc `ArgumentError` constructed on the spot would be more communicative.